src/NeoSharp.Core/Blockchain/Blockchain.cs

```diff
@@ -109,93 +119,20 @@ static int TransactionComparer(Stamp<Transaction> a, Stamp<Transaction> b)
         #region Blocks
 
         /// <inheritdoc />
-        public Task<bool> AddBlock(Block block)
+        public void AddBlock(Block block)
         {
-            if (block.Hash == null)
-            {
-                block.UpdateHash(_serializer, _crypto);
-            }
+            if (block == null) throw new ArgumentNullException(nameof(block));
 
-            if (CurrentBlock == null)
-            {
-                // Genesis
-
-                BlockPool.Push(block);
-
-                return PersistBlock();
-            }
-
-            // Small verification
-
-            if (block.Timestamp <= CurrentBlock.Timestamp ||
-                block.Index <= CurrentBlock.Index)
-            {
-                return Task.FromResult(false);
-            }
-
-            return Task.FromResult(BlockPool.Push(block));
+            _blockProcessor.AddBlock(block);
         }
 
         /// <inheritdoc />
-        public async Task<bool> PersistBlock()
+        public Task<bool> ContainsBlock(UInt256 blockHash)
```

Comment:
This method is not necessary. You can just call the ContainsBlock from anywhere in the BlockChain class.
Even the validations can be added to the BlockProcessor.ContainsBlock.